MatrixSDK/MXSession.m

```diff
@@ -4094,18 +4091,20 @@ - (MXHTTPOperation *)refreshHomeserverWellknown:(void (^)(MXWellKnown *))success
     MXLogDebug(@"[MXSession] refreshHomeserverWellknown");
     if (!autoDiscovery)
     {
-        NSString *homeServer;
-        
-        // Retrieve the domain from the user id as it can be different from the `MXRestClient.homeserver` that uses the client-server API endpoint domain.
-        NSString *userDomain = [MXTools serverNameInMatrixIdentifier:self.myUserId];
-        
-        if (userDomain)
-        {
-            homeServer =  [NSString stringWithFormat:@"https://%@", userDomain];
-        }
-        else
+        NSString *homeServer = [MXSDKOptions sharedInstance].wellknownDomainUrl;
```

Comment:
When the autoDiscovery is already instantiated, don't you think we should compare its url with the potential `wellknownDomainUrl` from MXKSDKOptions, in order to create it again if the urls don't match

Reply:
SDK options are build settings. We must not need to manage complex logic.